wpilibc/src/main/native/cpp/commands/Scheduler.cpp

```diff
@@ -9,84 +9,122 @@
 
 #include <algorithm>
 #include <set>
+#include <string>
+#include <vector>
 
 #include <hal/HAL.h>
+#include <networktables/NetworkTableEntry.h>
+#include <wpi/mutex.h>
 
 #include "frc/WPIErrors.h"
 #include "frc/buttons/ButtonScheduler.h"
+#include "frc/commands/Command.h"
 #include "frc/commands/Subsystem.h"
 #include "frc/smartdashboard/SendableBuilder.h"
 
 using namespace frc;
 
+struct Scheduler::Impl {
```

Comment:
The struct having member functions feels weird to me. It implies that there are invariants that need to be maintained, but having public member variables makes that unenforceable.

Reply:
We could make it a class with everything public if it's a style thing.  But almost everything needs to be public.

Reply:
Struct is the correct thing to use for aggregate types (which this is). What we have now is probably the least of the other evils.